build.psm1

```diff
@@ -2323,78 +2323,79 @@ function Start-CrossGen {
             [ValidateNotNullOrEmpty()]
             [String]
-            $CrossgenPath
+            $CrossgenPath,
+
+            [Parameter(Mandatory = $true)]
+            [ValidateSet("alpine-x64",
+                "linux-arm",
+                "linux-arm64",
+                "linux-x64",
+                "osx-arm64",
+                "osx-x64",
+                "win-arm",
+                "win-arm64",
+                "win7-x64",
+                "win7-x86")]
+            [string]
+            $Runtime
         )
 
+        $platformAssembliesPath = Split-Path $AssemblyPath[0] -Parent
 
-        $platformAssembliesPath = Split-Path $AssemblyPath -Parent
-        $crossgenFolder = Split-Path $CrossgenPath
-
-        try {
-            Push-Location $crossgenFolder
-
-            $symbolsPath = [System.IO.Path]::ChangeExtension($assemblyPath, ".pdb")
-
-            $createSymbolOptionName = $null
-            if($Environment.IsWindows)
-            {
-                $createSymbolOptionName = '-CreatePDB'
+        $targetOS, $targetArch = $Runtime -split '-'
 
+        # Special cases where OS / Arch does not conform with runtime names
+        switch ($Runtime) {
+            'alpine-x64' {
+                $targetOS = 'linux'
+                $targetArch = 'x64'
+             }
+            'win-arm' {
+                $targetOS = 'windows'
+                $targetArch = 'arm'
             }
-            elseif ($Environment.IsLinux)
-            {
-                $createSymbolOptionName = '-CreatePerfMap'
+            'win-arm64' {
+                $targetOS = 'windows'
+                $targetArch = 'arm64'
             }
-
-            if($createSymbolOptionName)
-            {
-                Start-NativeExecution {
-                    & $CrossgenPath -readytorun -platform_assemblies_paths $platformAssembliesPath $createSymbolOptionName $platformAssembliesPath $AssemblyPath
-                } | Write-Verbose
+            'win7-x64' {
+                $targetOS = 'windows'
+                $targetArch = 'x64'
+            }
+            'win7-x86' {
+                $targetOS = 'windows'
+                $targetArch = 'x86'
             }
+        }
 
-            # Rename the corresponding ni.dll assembly to be the same as the IL assembly
-            $niSymbolsPath = [System.IO.Path]::ChangeExtension($symbolsPath, "ni.pdb")
-            Rename-Item $niSymbolsPath $symbolsPath -Force -ErrorAction Stop
-        } finally {
-            Pop-Location
+        # The path to folder must end with directory separator
+        $dirSep = [System.IO.Path]::DirectorySeparatorChar
+        $platformAssembliesPath = if (-not $platformAssembliesPath.EndsWith($dirSep)) { $platformAssembliesPath + $dirSep }
+
+        Start-NativeExecution {
+            $crossgen2Params = @(
+                "-r"
+                $platformAssembliesPath
+                "--out-near-input"
+                "--single-file-compilation"
```

Comment:
Do we switch to single file?

Reply:
This is for crossgen'ing the input files individually. Not related to PublishSingleFile